src/test/scala/edu/ie3/simona/model/participant/HpModelSpec.scala

```diff
@@ -232,55 +232,209 @@ class HpModelSpec
       }
     }
 
-    "determining the flexibility options" when {
-      "the house is heated up and storage has space" should {
-        "deliver positive flexibility" in {
-          val house = thermalHouse(18, 22)
-            .copy(ethLosses = WattsPerKelvin(200))
-          val grid = thermalGrid(house, Some(thermalStorage))
-          val hp = hpModel(grid)
-          // Tick, at which the house is heated up
-          val relevantData = hpData.copy(currentTick = 2763L)
-          val thermalState = ThermalGridState(
-            Some(
-              ThermalHouseState(
-                0L,
-                Celsius(21),
-                Kilowatts(80),
-              )
+    "determining the flexibility options for different states" should {
+      "deliver correct flexibility options" in {
+        val testCases
+            : TableFor3[ThermalGridState, HpState, (Double, Double, Double)] =
```

Comment:
Nothing important, but question from my side: Does the specification of the type `TableFor3` provide any benefits?

Reply:
I don't think so. Might be a copy paste think, since we have TableFor2 / 3 / 4 ...
But the type annotations have no auto complete since for Table one need to provide them as string. 
```
        val testCases =
          Table(
            ("thermalState: ThermalGridState", "lastState: HpState", "expectedValues: (Double, Double, Double)"),
            ```

Reply:
Just removed the type specs and it works, let me know what you think

Reply:
Yes it works, but if you want to have the types you have to write them manually in Intellij. But I'm fine with this very little and only downside (maybe it's still possible and it's on me).

Reply:
Here, the types are all defined within the `forAll` clause anyway, maybe that suffices?